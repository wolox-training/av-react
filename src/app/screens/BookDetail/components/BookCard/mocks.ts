export const OK_BOOK_RESPONSE_API = {
  id: 1,
  author: 'author',
  title: 'title',
  imageUrl: 'imageurl',
  editor: 'editor',
  year: 1900,
  genre: 'genre',
  currentRent: false
};
