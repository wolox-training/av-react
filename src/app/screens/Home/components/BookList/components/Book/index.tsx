import React from 'react';
import i18next from 'i18next';

import { Books } from '~utils/types';

import styles from './styles.module.scss';

interface BookProps {
  books: Books[];
}

export default function Book({ books }: BookProps) {
  return (
    <>
      {books.map(book => (
        <div key={book.id} className={`${styles.book} column`}>
          <img
            src={book.imageUrl}
            alt={i18next.t('BooksList:bookAlt') as string}
            className={styles.bookImage}
          />
          <div className={styles.bookTitle}>{book.title}</div>
          <div className={styles.bookAuthor}>{book.author}</div>
        </div>
      ))}
    </>
  );
}
