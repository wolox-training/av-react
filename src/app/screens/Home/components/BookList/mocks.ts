export const OK_BOOKS_API_RESPONSE = {
  count: 1,
  currentPage: 1,
  nextPage: 1,
  page: [
    {
      id: 1,
      author: 'test author',
      title: 'test title',
      imageUrl: 'test url',
      publisher: 'test publisher',
      year: 1960,
      genre: 'test genre',
      createdAt: '2020-10-10',
      updatedAt: '2020-10-10'
    },
    {
      id: 2,
      author: 'test author 2',
      title: 'test title 2',
      imageUrl: 'test url 2',
      publisher: 'test publisher 2',
      year: 1960,
      genre: 'test genre 2',
      createdAt: '2020-10-10',
      updatedAt: '2020-10-10'
    }
  ],
  totalCount: 1,
  totalPages: 1
};
