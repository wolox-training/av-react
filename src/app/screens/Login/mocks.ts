export const OK_LOGIN_API_RESPONSE = {
  data: {
    'access-token': 'test',
    data: {
      email: 'test@test.com',
      firstName: 'test',
      id: 1,
      lastName: 'test',
      locale: 'es'
    }
  }
};
