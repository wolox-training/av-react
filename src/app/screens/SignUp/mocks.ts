export const OK_SIGNUP_API_RESPONSE = {
  headers: {
    'access-token': 'test'
  },
  data: {
    email: 'test@test.com',
    firstName: 'test',
    id: 1,
    lastName: 'test',
    locale: 'es'
  },
  ok: true,
  originalError: null,
  problem: null,
  status: 201
};
