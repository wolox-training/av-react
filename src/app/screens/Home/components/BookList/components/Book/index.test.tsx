import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { OK_BOOKS_API_RESPONSE } from '../../mocks';

import Book from './';

const MAX_BOOKS_LENGTH = 2;

describe('Book test', () => {
  test('Should match snapshot', () => {
    const { container } = render(
      <Router>
        <Book books={OK_BOOKS_API_RESPONSE.page} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  test('Should render books data', async () => {
    render(
      <Router>
        <Book books={OK_BOOKS_API_RESPONSE.page} />
      </Router>
    );
    await waitFor(() => {
      expect(screen.getAllByAltText('BooksList:bookAlt').length).toBe(MAX_BOOKS_LENGTH);
      expect(screen.getByText(OK_BOOKS_API_RESPONSE.page[0].title)).toBeInTheDocument();
      expect(screen.getByText(OK_BOOKS_API_RESPONSE.page[0].author)).toBeInTheDocument();
    });
  });
});
