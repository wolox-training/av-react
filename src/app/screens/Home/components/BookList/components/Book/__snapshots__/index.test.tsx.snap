// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book test Should match snapshot 1`] = `
<div>
  <div
    class="book column"
  >
    <img
      alt="BooksList:bookAlt"
      class="bookImage"
      src="test url"
    />
    <div
      class="bookTitle"
    >
      test title
    </div>
    <div
      class="bookAuthor"
    >
      test author
    </div>
  </div>
  <div
    class="book column"
  >
    <img
      alt="BooksList:bookAlt"
      class="bookImage"
      src="test url 2"
    />
    <div
      class="bookTitle"
    >
      test title 2
    </div>
    <div
      class="bookAuthor"
    >
      test author 2
    </div>
  </div>
</div>
`;
